jina/executors/indexers/vector.py

```diff
@@ -263,7 +267,7 @@ def query(self, keys: 'np.ndarray', top_k: int, *args, **kwargs) -> Tuple['np.nd
         else:
             raise NotImplementedError(f'{self.metric} is not implemented')
 
-        idx = dist.argsort(axis=1)[:, :top_k]
+        idx = np.argpartition(dist, kth=top_k, axis=1)[:, :top_k]
```

Comment:
You could also first do the `argpartition` and then the `argsort`. This does not break anything, but is still faster, since you only sort the `top_k` and not the whole list.

Reply:
Smart move... did not occur to me!

Reply:
or now that we are doing `batching` in the distance computation, we could maybe have some heap sort logic and do the top_k selection on the fly @hanxiao ?

Reply:
looks good. but by benchmarking, the improvement is minor, 2% on avg.

time is measured as the sum of 5x 100-query
```
old, topk=256
    [7.548 secs]
old, topk=512
    [7.585 secs]
this PR, topk=256
    [7.367 secs]
old, topk=512
    [7.428 secs]
```

Reply:
This also only start to be an issue, when the index becomes really large, not? The speed improvements depends upon the tail of the list and not upon the head.

Reply:
no significant difference